PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/editors/rediscache/RedisExplorerEditor.java

```diff
@@ -0,0 +1,204 @@
+        
+        cbDatabase = new Combo(cmpoMain, SWT.NONE);
+        cbDatabase.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, true, false, 1, 1));
+        
+        sashForm = new SashForm(cmpoMain, SWT.HORIZONTAL);
+        sashForm.setLayoutData(new GridData(SWT.FILL, SWT.FILL, false, true, 2, 1));
+        
+        cmpoKeyArea = new Composite(sashForm, SWT.BORDER);
+        cmpoKeyArea.setLayout(new GridLayout(2, false));
+        
+        txtKeySearchPattern = new Text(cmpoKeyArea, SWT.BORDER);
+        txtKeySearchPattern.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false, 1, 1));
+        
+        btnKeySearch = new Button(cmpoKeyArea, SWT.NONE);
+        btnKeySearch.setText("Search");
+        
+        table =  new Table(cmpoKeyArea, SWT.BORDER | SWT.SINGLE | SWT.V_SCROLL | SWT.H_SCROLL);
+        table.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, 2, 1));
+        table.setLinesVisible(true);
+        
+        //TODO: Remove this!
+        fillingTestData();
+        
+        btnAddKey = new Button(cmpoKeyArea, SWT.NONE);
+        btnAddKey.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, false, false, 2, 1));
+        btnAddKey.setText("Add");
+        
+        cmpoValueArea = new Composite(sashForm, SWT.BORDER);
+        cmpoValueArea.setLayout(new GridLayout(1, false));
+        sashForm.setWeights(new int[] {1, 1});
+        scrolledComposite.setContent(cmpoMain);
+        scrolledComposite.setMinSize(cmpoMain.computeSize(SWT.DEFAULT, SWT.DEFAULT));
+    }
+    
+    private void fillingTestData() {
+        //TODO: only for test!
+        TableItem stringItem = new TableItem(table, SWT.NONE);
+        stringItem.setText(0, "string");
+        
+        TableItem listItem = new TableItem(table, SWT.NONE);
+        listItem.setText(0, "list");
+        
+        TableItem setItem = new TableItem(table, SWT.NONE);
+        setItem.setText(0, "set");
+        
+        TableItem zsetItem = new TableItem(table, SWT.NONE);
+        zsetItem.setText(0, "zset");
+        
+        TableItem hashItem = new TableItem(table, SWT.NONE);
+        hashItem.setText(0, "hash");
+    }
+
+    @Override
+    public void onReadRedisDatabaseNum(String sid, String id) {
+        this.redisExplorerPresenter.onReadDbNum(sid, id);
+    }
+    
+    @Override
+    public void renderDbCombo(int num) {
+        cbDatabase.add(String.valueOf(num));
+        cbDatabase.select(0);
+    }
+    
+    @Override
+    public void setFocus() {
+        // Set the focus
+    }
+
+    @Override
+    public void doSave(IProgressMonitor monitor) {
+        // Do the Save operation
+    }
+
+    @Override
+    public void doSaveAs() {
+        // Do the Save As operation
+    }
+    
+    @Override
+    public void init(IEditorSite site, IEditorInput input) throws PartInitException {
+        setSite(site);
+        setInput(input);
+        RedisExplorerEditorInput redisInput = (RedisExplorerEditorInput) input;
```

Comment:
take care of this cast. Since init is public, we cannot make sure the input is RedisExplorerEditorInput